proof/dleq.go

```diff
@@ -0,0 +1,163 @@
+package proof
+
+import (
+	"errors"
+
+	"github.com/dedis/crypto/abstract"
+	"github.com/dedis/crypto/hash"
+	"github.com/dedis/crypto/random"
+)
+
+// This package provides functionality to create and verify non-interactive
+// zero-knowledge (NIZK) proofs for the equality (EQ) of discrete logarithms (DL).
+
+// DLEQ resembles a NIZK dlog-equality proof. Allows to handle multiple proofs.
+type DLEQ struct {
+	suite abstract.Suite
+	Base  []DLEQBase
+	Core  []DLEQCore
+}
+
+// DLEQBase contains the base points against which the core proof is created/verified.
+type DLEQBase struct {
+	g abstract.Point
+	h abstract.Point
+}
+
+// DLEQCore contains the core elements of the NIZK dlog-equality proof.
+type DLEQCore struct {
+	C  abstract.Scalar // challenge
```

Comment:
I don't really mind, I like when it's clean and neat, and if the comment can fit in one line that's good for me. 
At the same time, when you have struct with some long comments and some short comments, using both styles is bad style. 
What about something that reconciliate everyone ?
 - if all comments can fit in one line, then write it in one line
 - if one comments can't fit in one line, write all comments above

Otherwise, if we want to stick to one general rule, I'd prefer also on top so it avoids conflicts between long and short comments.